mmdet/core/visualization/image.py

```diff
@@ -88,29 +89,11 @@ def imshow_det_bboxes(img,
         if segms is not None:
             segms = segms[inds, ...]
 
-    mask_colors = []
-    if labels.shape[0] > 0:
-        if mask_color is None:
-            # Get random state before set seed, and restore random state later.
-            # Prevent loss of randomness.
-            # See: https://github.com/open-mmlab/mmdetection/issues/5844
-            state = np.random.get_state()
-            # random color
-            np.random.seed(42)
-            mask_colors = [
-                np.random.randint(0, 256, (1, 3), dtype=np.uint8)
-                for _ in range(max(labels) + 1)
-            ]
-            np.random.set_state(state)
-        else:
-            # specify  color
-            mask_colors = [
-                np.array(mmcv.color_val(mask_color)[::-1], dtype=np.uint8)
-            ] * (
-                max(labels) + 1)
-
-    bbox_color = color_val_matplotlib(bbox_color)
-    text_color = color_val_matplotlib(text_color)
+    max_label = int(max(labels)) if labels.shape[0] > 0 else -1
+    bbox_color = palette_val(get_palette(bbox_color, max_label + 1))
+    text_color = palette_val(get_palette(text_color, max_label + 1))
+    mask_color = get_palette(mask_color, max_label + 1)
```

Comment:
why this line does not use palette_val()?

Reply:
The mask_color in imshow_det_bboxes needs to be in range 0~255,  so I don't pass it through palette_val.

Reply:
Ok, I see.